src/services/matchmaking/matchmaking_service.cpp

```diff
@@ -113,4 +138,162 @@ namespace big
+	{
+		patch_matchmaking_attributes(attributes);
+
+		if (!g.spoofing.multiplex_session)
+			return false;
+		
+		if (status->status)
+			return true;
+
+		status->status = 1; // set in progress
+
+		// create the first advertisement
+		g_fiber_pool->queue_job([this, num_slots, available_slots, info, attributes, out_id, status] {
+			rage::rlTaskStatus our_status{};
+			if (!g_hooking->get_original<hooks::advertise_session>()(0, num_slots, available_slots, attributes, -1, info, out_id, &our_status))
+			{
+				LOG(WARNING) << __FUNCTION__ ": advertise_session returned false for first advertisement";
+				status->status = 2;
+				return;
+			}
+
+			while (our_status.status == 1)
+				script::get_current()->yield();
+
+			if (our_status.status == 2)
+			{
+				LOG(WARNING) << __FUNCTION__ ": advertise_session failed for first advertisement";
+				status->status = 2;
+				return;
+			}
+
+			MatchmakingId primary_id = *out_id; // create a copy if the original memory gets deallocated
+			std::uint32_t id_hash             = rage::joaat(primary_id.m_data1);
+
+			// m_data1 is generated from m_data2 and m_data3
+			m_multiplexed_sessions.emplace(id_hash, std::vector<MatchmakingId>{ });
+
+			// create multiplex advertisements
+			for (int i = 0; i < (g.spoofing.multiplex_count - 1); i++)
+			{
+				g_fiber_pool->queue_job([this, num_slots, available_slots, info, attributes, id_hash, i] {
+					rage::rlTaskStatus status;
+					MatchmakingId multiplexed_id;
+					if (!g_hooking->get_original<hooks::advertise_session>()(0, num_slots, available_slots, attributes, -1, info, &multiplexed_id, &status))
+					{
+						LOG(WARNING) << __FUNCTION__ ": advertise_session returned false for multiplex task " << i;
+						return;
+					}
+					
+					while (status.status == 1)
+						script::get_current()->yield();
+
+					if (status.status == 2)
+					{
+						LOG(WARNING) << __FUNCTION__ ": advertise_session failed for multiplex task " << i;
+						return;
+					}
+
+					if (auto it = m_multiplexed_sessions.find(id_hash); it != m_multiplexed_sessions.end())
+					{
+						it->second.push_back(multiplexed_id);
+					}
+					else
+					{
+						LOG(WARNING) << __FUNCTION__ ": created a multiplexed session advertisement, but the primary advertisement no longer exists";
+					}
+				});
+			}
+			status->status = 3; // return success for original caller
+		});
+
+		return true;
+	}
+
+	void matchmaking_service::handle_update(int num_slots, int available_slots, rage::rlSessionInfo* info, MatchmakingAttributes* attributes, MatchmakingId* id)
+	{
+		patch_matchmaking_attributes(attributes);
+
+		// this can be fire and forget, but it's probably a good idea to be notified if something goes wrong
+		if (auto it = m_multiplexed_sessions.find(rage::joaat(id->m_data1)); it != m_multiplexed_sessions.end())
+		{
+			if (!g.spoofing.multiplex_session)
+			{
+				// option disabled mid-session
+				return;
+			}
+
+			int i = 0;

```

Comment:
Static? Otherwise the compiler will discard this after it falls out of scope.

Reply:
But the lambda is executed asynchronously. You may want to consider using atomic_bool or something, otherwise you may have something like 

```
update_session_advertisement returned false for multiplex task 3
update_session_advertisement returned false for multiplex task 4
update_session_advertisement returned false for multiplex task 5
update_session_advertisement returned false for multiplex task 4
update_session_advertisement returned false for multiplex task 5
```